site/en/docs/android/trusted-web-activity/whats-new/index.md

```diff
@@ -0,0 +1,429 @@
+      // is activated.
+      return await caches.match(OFFLINE_PAGE);
+    }
+  })());
+});
+
+async function fetchAndHandleError(request) {
+  const cache = await caches.open(RUNTIME_CACHE);
+  const response = await fetch(request);
+
+  // Throw an error if the response returns one of the status
+  // that trigger the Quality Criteria.
+  if (response.status === 404 ||
+      response.status >= 500 && response.status < 600) {
+    throw new Error(`Server responded with status: ${response.status}`);
+  }
+
+  // Cache the response if the request is successful.
+  cache.put(request, response.clone());
+  return response;
+}
+```
+
+[Workbox][9] bakes in best practices and removes boilerplate when using service workers.
+Alternatively, consider using a Workbox plugin to handle those scenarios:
+
+```javascript
+export class FallbackOnErrorPlugin {
+  constructor(offlineFallbackUrl, notFoundFallbackUrl, serverErrorFallbackUrl) {
+    this.notFoundFallbackUrl = notFoundFallbackUrl;
+    this.offlineFallbackUrl = offlineFallbackUrl;
+    this.serverErrorFallbackUrl = serverErrorFallbackUrl;
+  }
+
+  checkTrustedWebActivityCrash(response) {
+    if (response.status === 404 || response.status >= 500 && response.status <= 600) {
+      const type = response.status === 404 ? 'E_NOT_FOUND' : 'E_SERVER_ERROR';
+      const error = new Error(`Invalid response status (${response.status})`);
+      error.type = type;
+      throw error;
+    }
+  }
+
+  // This is called whenever there's a network response,
+  // but we want special behavior for 404 and 5**.
+  fetchDidSucceed({response}) {
+    // Cause a crash if this is a Trusted Web Activity crash.
+    this.checkTrustedWebActivityCrash(response);
+
+    // If it's a good response, it can be used as-is.
+    return response;
+  }
+
+  // This callback is new in Workbox v6, and is triggered whenever
+  // an error (including a NetworkError) is thrown when a handler runs.
+  handlerDidError(details) {
+    let fallbackURL;
+    switch (details.error.details.error.type) {
+      case 'E_NOT_FOUND': fallbackURL = this.notFoundFallbackUrl; break;
+      case 'E_SERVER_ERROR': fallbackURL = this.serverErrorFallbackUrl; break;
+      default: fallbackURL = this.offlineFallbackUrl;
+    }
+
+    return caches.match(fallbackURL, {
+      // Use ignoreSearch as a shortcut to work with precached URLs
+      // that have _WB_REVISION parameters.
+      ignoreSearch: true,
+    });
+  }
+}
+```
+
+## Google Play Billing
+
+Besides allowing your app to sell digital goods and subscriptions on the Play Store,
+[Google Play Billing][10] offers tools for managing your catalog, prices and subscriptions, useful
+reports, and a checkout flow powered by the Play Store that is already familiar to your users. It
+is also a requirement for applications published on the Play Store that sell digital goods.
+
+Chrome 88 will launch with an origin trial on Android that enables the integration of
```

Comment:
Optional but you could tag this post with `chrome-88` or `chrome-89` (or both!) to have it show up adjacent to other content for that release.

Reply:
Yeah, I considered that, but since the post doesn't have a hero image, it breaks the layout in the tag page. I was also wondering about moving this to the blog instead of docs, but ran into the same problem. It does sound to me the best approach would be to create a hero image for this.